assists/openamp-xlnx.py

```diff
@@ -75,6 +122,19 @@ def parse_ipis_for_rpu(sdt, domain_node, rpu_cpu_node, options):
     rpu_cpu_node + ipis_prop
     return ipi_list
 
+def parse_memory_carevouts_for_rpu(sdt, domain_node, rpu_cpu_node, options):
+    try:
+        verbose = options['verbose']
+    except:
+        verbose = 0
+
+    carveout_list = []
+    for k,v in sdt.tree.__nodes__.items():
```

Comment:
In theory, we shouldn't be going directly as the __nodes__ member, since the implementation of it can change in the future (I'd have made it private if that was possible) ...

Have you tried this with the standard iterator ?

i.e.:

for node in sdt.tree:
   if channel in node.abs_path and "openamp,xlnx,mem-carveout" in node['compatible].value[0]:

... etc.

Reply:
ok this is now done as you suggested in later commit thank you